S2VX.Game/Editor/SelectToolState.cs

```diff
@@ -1,5 +1,146 @@
-﻿namespace S2VX.Game.Editor {
+﻿using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Events;
+using osuTK;
+using osuTK.Graphics;
+using S2VX.Game.Story;
+using System;
+using System.Collections.Generic;
+
+namespace S2VX.Game.Editor {
     public class SelectToolState : ToolState {
+        [Resolved]
+        private S2VXStory Story { get; set; } = null;
+        [Resolved]
+        private S2VXEditor Editor { get; set; } = null;
+        private Dictionary<Note, double> SelectedNoteToTime { get; set; } = new Dictionary<Note, double>();
+        private Dictionary<Drawable, Note> NoteSelectionToNote { get; set; } = new Dictionary<Drawable, Note>();
+
+        private const float SelectionIndicatorThickness = 0.025f;
+
+        private bool DragTimelineNote { get; set; }
+        private Dictionary<Note, double> NoteToDragPointDelta { get; set; } = new Dictionary<Note, double>();
+
+        private static bool MouseIsOnNote(Vector2 mousePos, RelativeBox timelineNote) {
+            var leftBound = timelineNote.DrawPosition.X - timelineNote.DrawSize.X / 2;
+            var rightBound = timelineNote.DrawPosition.X + timelineNote.DrawSize.X / 2;
+            var topBound = timelineNote.DrawPosition.Y - timelineNote.DrawSize.Y / 2;
+            var bottomBound = timelineNote.DrawPosition.Y + timelineNote.DrawSize.Y / 2;
+
+            var mouseInXRange = leftBound <= mousePos.X && mousePos.X <= rightBound;
+            var mouseInYRange = topBound <= mousePos.Y && mousePos.Y <= bottomBound;
+            return mouseInXRange && mouseInYRange;
+        }
+
+        public override bool OnToolMouseDown(MouseDownEvent _) {
```

Comment:
Don't use _ as a variable name. _ is supposed to indicate that a particular variable is unused. In this case, just naming the MouseDownEvent a simple `event` or `e` (I kind of prefer this last one) is sufficient. See this doc for more details regarding discards: https://docs.microsoft.com/en-us/dotnet/csharp/discards

Reply:
Ok I'll change the _ to e